src/platform/packages/shared/kbn-scout/src/config/discovery/search_configs.ts

```diff
@@ -13,35 +13,78 @@ import { ToolingLog } from '@kbn/tooling-log';
 
 export const DEFAULT_TEST_PATH_PATTERNS = ['src/platform/plugins', 'x-pack/**/plugins'];
 
+interface PluginScoutConfig {
+  group: string;
+  pluginPath: string;
+  usesParallelWorkers: boolean;
+  configs: string[];
+}
+
 export const getScoutPlaywrightConfigs = (searchPaths: string[], log: ToolingLog) => {
   const patterns = searchPaths.map((basePath) =>
     path.join(basePath, '**/ui_tests/{playwright.config.ts,parallel.playwright.config.ts}')
   );
 
-  log.info('Searching for playwright config files in the following paths:');
+  log.info('Searching for Playwright config files in the following paths:');
   patterns.forEach((pattern) => log.info(`- ${pattern}`));
   log.info(''); // Add a newline for better readability
 
   const files = patterns.flatMap((pattern) => fastGlob.sync(pattern, { onlyFiles: true }));
 
-  // Group config files by plugin
-  const plugins = files.reduce((acc: Map<string, string[]>, filePath: string) => {
-    const match = filePath.match(
-      /(?:src\/platform\/plugins|x-pack\/.*?\/plugins)\/(?:.*?\/)?([^\/]+)\/ui_tests\//
-    );
-    const pluginName = match ? match[1] : null;
+  const typeMappings: Record<string, string> = {
+    'x-pack/solutions/security': 'security',
+    'x-pack/solutions/search': 'search',
+    'x-pack/solutions/observability': 'observability',
+    'x-pack/platform/plugins': 'platform',
+    'src/platform/plugins': 'platform',
+  };
+
+  const matchPluginPath = (filePath: string): { pluginPath: string; pluginName: string } | null => {
+    const regexes = [
+      /(x-pack\/platform\/plugins\/(?:private|shared|[^\/]+)\/([^\/]+))\/ui_tests\//,
+      /(x-pack\/solutions\/[^\/]+\/plugins\/([^\/]+))\/ui_tests\//,
```

Comment:
Did you intentionally exclude the solution specific `packages` directories here?
Similarly, would we want `packages` to be in `DEFAULT_TEST_PATH_PATTERNS`?

Reply:
Afaik we don't have Scout playwright configs in the `packages` only in `plugins/`, so I only included the paths where we expect them to be:

```
src/platform/plugins/...
x-pack/platform/plugins/...
x-pack/solution/../plugins/...
```

if we will decide to go with packages as well, we can later update it. WDYT?